src/sentry/static/sentry/app/locale.tsx

```diff
@@ -321,7 +321,14 @@ export function ngettext(singular: string, plural: string, ...args: FormatArg[])
 
   if (args.length > 0) {
     countArg = (args[0] as number) || 0;
-    args = [countArg.toLocaleString(), ...args.slice(1)];
+
+    // using %d may cause issues - for example if we do `.toLocaleString` on a number higher than 999
+    if ((singular + plural).includes('%d')) {
+      // eslint-disable-next-line no-console
+      console.error('You should not use %d within tn(), use %s instead');
```

Comment:
Let's make this an Error object so that you get a stack trace.

Reply:
Cool, I made it an Error object and updated a comment little to be more descriptive of why we are doing this. Also, here is a PR for lint rule: https://github.com/getsentry/eslint-config-sentry/pull/50